### PR TITLE
feature: filterable for adapter

### PR DESCRIPTION
**Pull request by MrBrowner**
added filter var to use search feature in adapter

### CODE REVIEW

File: adapt/src/main/java/io/github/vshnv/adapt/adapter/AdaptAdapter.kt [hunk 1]
```diff
@@ -1,11 +1,15 @@
 package io.github.vshnv.adapt.adapter
 
+import android.widget.Filter
+import android.widget.Filterable
 import androidx.recyclerview.widget.RecyclerView
 
-abstract class AdaptAdapter<T>: RecyclerView.Adapter<AdaptViewHolder<T>>() {
+abstract class AdaptAdapter<T>: RecyclerView.Adapter<AdaptViewHolder<T>>(), Filterable {
     abstract val currentList: List<T>
+    abstract var searchFilterable: Filter?
```
Comment (Vshnv) [inline adapt/src/main/java/io/github/vshnv/adapt/adapter/AdaptAdapter.kt]: This shouldn't ideally be exposed. 

Ideally we should try to implement the `Filter` as a DSL into AdaptScope and hide the instance in the implementation.

File: adapt/src/main/java/io/github/vshnv/adapt/adapter/LifecycleAwareAdaptAdapter.kt [hunk 2]
```diff
@@ -28,12 +29,22 @@ class LifecycleAwareAdaptAdapter<T : Any>(private val viewTypeMapper: ((T, Int)
     private val mDiffer: AsyncListDiffer<T> = AsyncListDiffer(this, diffCallback)
     override val currentList: List<T>
         get() = mDiffer.currentList
+    private var fullList: MutableList<T> = mutableListOf()
```
Comment (Vshnv) [inline adapt/src/main/java/io/github/vshnv/adapt/adapter/LifecycleAwareAdaptAdapter.kt]: Storing the entire list and passing the actual list to the differ might not work in this case. submitData would reset fulldata so theres no way to notify the adapter of the filtered list. Might be worth it to extend the class and provide a filtering option there with explicit note of "unfilteredData" and "data".

File: adapt/src/main/java/io/github/vshnv/adapt/adapter/AdaptAdapter.kt (hunk 1 above)
Comment (MrBrowner) [reply adapt/src/main/java/io/github/vshnv/adapt/adapter/AdaptAdapter.kt]: I think you can check my other branch "**i1**" in fork repo.

there i already added the filter in AdaptScope

File: adapt/src/main/java/io/github/vshnv/adapt/adapter/LifecycleAwareAdaptAdapter.kt (hunk 2 above)
Comment (MrBrowner) [reply adapt/src/main/java/io/github/vshnv/adapt/adapter/LifecycleAwareAdaptAdapter.kt]: I didn't get what actually you are talking about!  🥲 ..

**Pull request closed without merging**